src/components/AddItem.jsx

```diff
@@ -1,7 +1,102 @@
-import React from 'react';
+import React, { useState } from 'react';
+import { collection, addDoc } from 'firebase/firestore';
+import { db } from '../lib/firebase';
+
+const radioButtonOptions = [
+  {
+    option: 'soon',
+    value: 7,
+    defaultChecked: true,
+  },
+  {
+    option: 'kind of soon',
+    value: 14,
+    defaultChecked: false,
+  },
+  {
+    option: 'not soon',
+    value: 30,
+    defaultChecked: false,
+  },
+];
 
 function AddItem() {
-  return <div>Add Item</div>;
+  const [itemName, setItemName] = useState('');
+  const [frequency, setFrequency] = useState(7);
+  const [notification, setNotification] = useState('');
+
+  //retrive the token from localStorage
+  const token = localStorage.getItem('list-token');
+
+  //handle submit for the form
+  const handleSubmit = async (e) => {
+    e.preventDefault();
+    //create new doc reference for text input and save to firestore database
+    //use a token to save new items into specific collection (shopping list)
+    try {
+      const docRef = await addDoc(collection(db, token), {
+        //data points being sent to firebase, object format
```

Comment:
Since we will need to associate items with a particular token/list, maybe we can create a new collection in Firestore called 'lists'. Each list can be associated with a specific token and then have a sub-collection of items. That might be something to take care of later down the road...